src/containers/DatasetsContainer.jsx

```diff
@@ -0,0 +1,167 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Route } from 'react-router';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import { fetchDataset, emptyDataset, createDataset, removeDataset, updateDataset }
+  from '../actions/dataset';
+import { emptyResource } from '../actions/resource';
+import DatasetList from '../components/DatasetList';
+import DatasetDetail from '../components/DatasetDetail/DatasetDetail';
+import ResourceDetail from '../components/ResourceDetail/ResourceDetail';
+
+import './datasets-container.scss';
+
+const mapStateToProps = state => ({
+  dataset: state.dataset,
+  datasets: state.datasets,
+  resource: state.resource,
+  schema: state.schema,
+  uiDataset: state.uiDataset,
+  uiResource: state.uiResource
+});
+
+const mapDispatchToProps = dispatch => bindActionCreators({
+  onFetch: fetchDataset,
+  onEmpty: emptyDataset,
+  onCreate: createDataset,
+  onRemove: removeDataset,
+  onUpdate: updateDataset,
+  onEmptyResource: emptyResource
+}, dispatch);
+
+class DatasetsContainer extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      resourceToDataset: props.resourceToDataset
+    };
+
+    this.handleResourceToDataset = this.handleResourceToDataset.bind(this);
+    this.emptyResourceToDataset = this.emptyResourceToDataset.bind(this);
+  }
+
+  handleResourceToDataset(resource) {
+    this.setState({
+      resourceToDataset: resource
+    });
+  }
+
+  emptyResourceToDataset() {
+    this.setState({
+      resourceToDataset: {}
+    });
+  }
+
+  render() {
+    return (
+      <section>
+        <Route
+          exact
+          path="/dcatd_admin/datasets"
+          render={() => (
+            <DatasetList
+              datasets={this.props.datasets}
+            />
+          )}
+        />
+        <Route
+          exact
+          path="/dcatd_admin/datasets/:id([\w-]{6,})"
+          render={() => (
```

Comment:
This looks like you could use a wrapper around all the components that you want to render per route.
I would maybe create a dir named "pages" and place all the wrapper components inside. Import those components in this file and add it to the correct <Route> components.